Backend/OnboardingInsuranceAPI/Areas/User/RegisterUserHandler.cs

```diff
@@ -17,12 +17,16 @@ public RegisterUserHandler(DataContext context, ILogger<RegisterUserHandler> log
         _log = log;
     }
 
-    public async Task CreateUser(string pid)
+    public async Task CreateUser(string pid, string email)
     {
         var user = await _context.Users.FirstOrDefaultAsync(x => x.Pid == pid);//check if user exist
         if (user == null)
```

Comment:
Try changing it so that you don't wrap the whole code into if. Rather have "if there is user return error" then "if there is no user create it".
It makes the code more readable.

Reply:
Its little bit different thing, this backend function check every time when logging its checking event sign-in or sign-up because on custom policies sign-in and sign-up as I understand is a one event, I can't catch sign-in or sign-up separately. So I have to call registration every time and when user logging, if pid present in db its do nothing, otherwise its create new user.